### PR TITLE
remove InsertBuffer

### PR DESCRIPTION
**Pull request by thorfour**
Insertion via Parquet files is now officially unsupported

### CODE REVIEW

**Review by brancz — APPROVED** [review]
🥳

File: logictest/runner.go [hunk 1]
```diff
@@ -255,24 +254,18 @@ func (r *Runner) handleInsert(ctx context.Context, c *datadriven.TestData, arrow
 
 	buf.Sort()
 
-	if arrow {
-		converter := pqarrow.NewParquetConverter(memory.NewGoAllocator(), logicalplan.IterOptions{})
-		defer converter.Close()
+	converter := pqarrow.NewParquetConverter(memory.NewGoAllocator(), logicalplan.IterOptions{})
```
Comment (asubiotto) [inline logictest/runner.go:257]: We're still creating records from parquet buffers here. Maybe change this to construct arrow records directly in this PR or leave that as a TODO.

File: compaction_test.go [hunk 2]
```diff
@@ -329,9 +310,7 @@ func TestCompaction(t *testing.T) {
 							"tests assume only a single granule as input",
 						)
 						cfg := table.db.columnStore.compactionConfig
-						if asArrow {
-							cfg.l1ToGranuleSizeRatio = 0.6 // use a different ratio for arrow records
-						}
+						cfg.l1ToGranuleSizeRatio = 0.6 // use a different ratio for arrow records
```
Comment (asubiotto) [inline compaction_test.go:313]: nit: is the comment still relevant?

**Review by asubiotto — APPROVED** [review]
Very nice to see this. Could you create an issue to revisit the `ParquetConverter` usages to see if we can create records directly?

File: logictest/runner.go (hunk 1 above)
Comment (thorfour) [reply logictest/runner.go:257]: Good call. I'll add an issue and leave it as todo

File: compaction_test.go (hunk 2 above)
Comment (thorfour) [reply compaction_test.go:313]: Kind of? The test needs that ratio but it isn't using a different one since now there's only the one

**Pull request merged**